APMrover2/GCS_Mavlink.cpp

```diff
@@ -746,6 +746,12 @@ MAV_RESULT GCS_MAVLINK_Rover::handle_command_long_packet(const mavlink_command_l
                                               static_cast<int16_t>(packet.param3),
                                               packet.param4);
 
+    case MAV_CMD_DO_ENGINE_CONTROL:
+        if (!rover.g2.ice_control.engine_control(packet.param1, packet.param2, packet.param3)) {
```

Comment:
normally we would call this function, "handle_msg" or "handle_engine_control_msg" or something like this

Reply:
.. but I realise this is an existing issue..

Reply:
.. ah, but it doesn't accept a mavlink message explicitely.. forget it sorry!